Fortran/UnitTests/finalization/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+add_executable( test_finalization
```

Comment:
This adds an executable, but doesn't actually run the test. You should add a reference output and some boilerplate similar to what's been done for [SNAP](https://github.com/llvm/llvm-test-suite/blob/edc9be64895f43c79aa34fad32801ada216ce229/Fortran/SNAP/src/CMakeLists.txt#L170). Note that that implementation is a bit messy at the moment - if you have some cycles to spend on this, you could try to extract some CMake helpers to use both here and in SNAP, similar to what we have for C++ tests.